### PR TITLE
Get players secure/fight/chase modes

### PR DESCRIPTION
**Pull request by Xagul**
hasSecureMode
getFightMode
getChaseMode

### CODE REVIEW

File: src/luascript.cpp [hunk 1]
```diff
@@ -2229,6 +2229,10 @@ void LuaScriptInterface::registerFunctions()
 	registerMethod("Player", "getContainerId", LuaScriptInterface::luaPlayerGetContainerId);
 	registerMethod("Player", "getContainerById", LuaScriptInterface::luaPlayerGetContainerById);
 	registerMethod("Player", "getContainerIndex", LuaScriptInterface::luaPlayerGetContainerIndex);
+
```
Comment (WibbenZ) [inline src/luascript.cpp]: Trailing

File: src/luascript.cpp [hunk 2]
```diff
@@ -9255,6 +9259,45 @@ int LuaScriptInterface::luaPlayerGetContainerIndex(lua_State* L)
...
 }
 
+int LuaScriptInterface::luaPlayerHasSecureMode(lua_State* L)
+{
+	// player:hasSecureMode()
+	Player* player = getUserdata<Player>(L, 1);
+	if (player) {
+		pushBoolean(L, player->hasSecureMode());
+	}
+	else {
```
Comment (WibbenZ) [inline src/luascript.cpp]: Indentation

File: src/luascript.cpp [hunk 3]
```diff
@@ -9255,6 +9259,45 @@ int LuaScriptInterface::luaPlayerGetContainerIndex(lua_State* L)
...
+}
+
+int LuaScriptInterface::luaPlayerGetFightMode(lua_State* L)
+{
+	// player:getFightMode()
+	Player* player = getUserdata<Player>(L, 1);
+	if (player) {
+		lua_pushnumber(L, player->fightMode);
+	}
+	else {
```
Comment (WibbenZ) [inline src/luascript.cpp]: Indentation

File: src/luascript.cpp [hunk 4]
```diff
@@ -9255,6 +9259,45 @@ int LuaScriptInterface::luaPlayerGetContainerIndex(lua_State* L)
...
+}
+
+int LuaScriptInterface::luaPlayerGetChaseMode(lua_State* L)
+{
+	// player:getChaseMode()
+	Player* player = getUserdata<Player>(L, 1);
+	if (player) {
+		lua_pushnumber(L, player->chaseMode);
+	}
+	else {
```
Comment (WibbenZ) [inline src/luascript.cpp]: Indentation

File: src/luascript.h [hunk 5]
```diff
@@ -970,6 +970,10 @@ class LuaScriptInterface
 		static int luaPlayerGetContainerId(lua_State* L);
 		static int luaPlayerGetContainerById(lua_State* L);
 		static int luaPlayerGetContainerIndex(lua_State* L);
+
```
Comment (WibbenZ) [inline src/luascript.h]: Trailing

**Comment by Schiffers** [comment]
what this functions do? can be used in spells or something like that?

**Comment by WibbenZ** [comment]
It can be used anywhere where you have player userdata.
So yes, even spells(as long as you use creature:getPlayer()).

**Comment by marksamman** [comment]
We should change chaseMode to a boolean and call the Lua function hasChaseMode instead.

**Comment by Schiffers** [comment]
I agree with Mark ..

**Pull request closed without merging**